build-system/runner/src/org/ampproject/AmpCommandLineRunner.java

```diff
@@ -68,6 +69,8 @@ protected AmpCommandLineRunner(String[] args) {
     }
     CompilerOptions options = super.createOptions();
     options.setCollapseProperties(true);
+    // Warn on improper uses of `this` that `collapseProperties` assumes.
+    options.setCheckGlobalThisLevel(CheckLevel.WARNING);
```

Comment:
Done. Note that this will only run for `gulp check-test` and will be an error rather than a warning.